src/server/pmix_server.c

```diff
@@ -264,17 +270,52 @@ PMIX_EXPORT pmix_status_t PMIx_server_init(pmix_server_module_t *module,
                     lt->owner = info[n].value.data.uint32;
                     lt->owner_given = true;
                 }
+                /* push this onto our protected list of keys not
+                 * to be passed to the clients */
+                pmix_argv_append_nosize(&protected, PMIX_USERID);
             } else if (0 == strcmp(info[n].key, PMIX_GRPID)) {
                /* the grpid is in the uint32_t storage */
                 PMIX_LIST_FOREACH(lt, &pmix_server_globals.listeners, pmix_listener_t) {
                     lt->group = info[n].value.data.uint32;
                     lt->group_given = true;
                 }
+                /* push this onto our protected list of keys not
+                 * to be passed to the clients */
+                pmix_argv_append_nosize(&protected, PMIX_GRPID);
             } else if (0 == strcmp(info[n].key, PMIX_SOCKET_MODE)) {
                 /* socket mode is in the uint32_t storage */
                 PMIX_LIST_FOREACH(lt, &pmix_server_globals.listeners, pmix_listener_t) {
                     lt->mode = info[n].value.data.uint32;
                 }
+            } else if (0 == strcmp(info[n].key, PMIX_SERVER_TOOL_SUPPORT)) {
+                pmix_listener_t *tl = PMIX_NEW(pmix_listener_t);
+                tl -> address.sun_family = AF_UNIX;
+                tl->protocol = PMIX_PROTOCOL_TOOL;
+                /* Get up to 10 chars of hostname.*/
+                gethostname(myhostname, myhostnamelen);
```

Comment:
@rhc54 you need this inline patch
```patch
diff --git a/opal/mca/pmix/pmix2x/pmix/src/server/pmix_server.c b/opal/mca/pmix/pmix2x/pmix/src/server/pmix_server.c
index 0533b38..18ac503 100644
--- a/opal/mca/pmix/pmix2x/pmix/src/server/pmix_server.c
+++ b/opal/mca/pmix/pmix2x/pmix/src/server/pmix_server.c
@@ -227,7 +227,7 @@ PMIX_EXPORT pmix_status_t PMIx_server_init(pmix_server_module_t *module,
     size_t n, m;
     pmix_kval_t kv;
     pmix_listener_t *lt;
-    int myhostnamelen = 10;
+    int myhostnamelen = 11;
     char myhostname[myhostnamelen];
     char *pmix_pid, *tdir;
     char **protected = NULL;
@@ -293,6 +293,10 @@ PMIX_EXPORT pmix_status_t PMIx_server_init(pmix_server_module_t *module,
                 tl->protocol = PMIX_PROTOCOL_TOOL;
                 /* Get up to 10 chars of hostname.*/
                 gethostname(myhostname, myhostnamelen);
+                /* gethostname might not null terminate myhostname
+                 * if hostname size is greater than myhostnamelen,
+                 * so do it ourself */
+                myhostname[10] = '\0';
                 /* need to put this in the global tmpdir as opposed to
                  * where the server tmpdir might be */
                 if (NULL == (tdir = getenv("TMPDIR"))) {
```

otherwise i get some random hangs on my centos 7 box with a long hostname